vms/avm/vm.go

```diff
@@ -495,26 +496,34 @@ func (vm *VM) GetUTXOs(
 		} else if comp == 0 {
 			start = startUTXOID
 		}
-		utxoIDs, err := vm.state.Funds(addr.Bytes(), start, limit) // Get UTXOs associated with [addr]
-		if err != nil {
-			return nil, ids.ShortID{}, ids.ID{}, fmt.Errorf("couldn't get UTXOs for address %s", addr)
-		}
-		for _, utxoID := range utxoIDs {
-			if seen.Contains(utxoID) { // Already have this UTXO in the list
-				continue
-			}
-			utxo, err := vm.state.UTXO(utxoID)
+		for {
+			currLimit := limit
```

Comment:
Do we want to reset `currLimit := limit` each loop?

For example, suppose `limit` is 10, and we get 9 UTXOs from the first address.
Then we get back to get back here and set `currLimit` to 10 and potentially return more than 10 UTXOs from this method even though `limit` is 10

Reply:
Yeah - let me review that ! 👌

Reply:
Yeah this methods logic is not easy to read - I'm happy to refactor it.

To address this scenario, it will still work.
If `limit = 10` and we have 9 UTXOs on the first address it will exit the loop later at `if paginate || len(utxoIDs) == 0` .
And will go on the next address. 

If the `paginate` is true it will continue to fetch all the utxos for that address.